Done.
Setting up the random seed for reproducibility...
Done.
class_name:  OracleTrainer
Initializing model...
Done.
Initializing datasets...
Done.
Initializing clients...
class_name:  OracleClient
clients_shared_args
Traceback (most recent call last):
  File "/home/utente/Scrivania/PROVA/LADD/src/run.py", line 49, in <module>
    run_experiment()
  File "/home/utente/Scrivania/PROVA/LADD/src/run.py", line 15, in run_experiment
    trainer = trainer_class(args, writer, device, rank, world_size)
  File "/home/utente/Scrivania/PROVA/LADD/src/federated/trainers/oracle_trainer.py", line 8, in __init__
    super().__init__(args, writer, device, rank, world_size)
  File "/home/utente/Scrivania/PROVA/LADD/src/federated/trainers/trainer.py", line 11, in __init__
    super().__init__(args, writer, device, rank, world_size)
  File "/home/utente/Scrivania/PROVA/LADD/src/general_trainer.py", line 58, in __init__
    self.__clients_setup()
  File "/home/utente/Scrivania/PROVA/LADD/src/general_trainer.py", line 143, in __clients_setup
    cl = client_class(**cl_args, batch_size=batch_size, test_user=split == 'test')
  File "/home/utente/Scrivania/PROVA/LADD/src/clients/oracle_client.py", line 16, in __init__
    super().__init__(args, client_id, dataset, model, writer, batch_size, world_size, rank, num_gpu,
  File "/home/utente/Scrivania/PROVA/LADD/src/clients/client.py", line 65, in __init__
    self.profiler_path = os.path.join('profiler', self.args.profiler_folder) if self.args.profiler_folder else None
KeyboardInterrupt
Error in sys.excepthook:
Traceback (most recent call last):
  File "/home/utente/pycharm/plugins/python-ce/helpers/pydev/_pydevd_bundle/pydevd_constants.py", line 58, in dummy_excepthook
    def dummy_excepthook(exctype, value, traceback):
KeyboardInterrupt
Original exception was:
Traceback (most recent call last):
  File "/home/utente/pycharm/plugins/python-ce/helpers/pydev/pydevd.py", line 2195, in <module>
    main()
  File "/home/utente/pycharm/plugins/python-ce/helpers/pydev/pydevd.py", line 2177, in main
    globals = debugger.run(setup['file'], None, None, is_module)
  File "/home/utente/pycharm/plugins/python-ce/helpers/pydev/pydevd.py", line 1489, in run
    return self._exec(is_module, entry_point_fn, module_name, file, globals, locals)
  File "/home/utente/pycharm/plugins/python-ce/helpers/pydev/pydevd.py", line 1496, in _exec
    pydev_imports.execfile(file, globals, locals)  # execute the script
  File "/home/utente/pycharm/plugins/python-ce/helpers/pydev/_pydev_imps/_pydev_execfile.py", line 18, in execfile
    exec(compile(contents+"\n", file, 'exec'), glob, loc)
  File "/home/utente/Scrivania/PROVA/LADD/src/run.py", line 49, in <module>
    run_experiment()
  File "/home/utente/Scrivania/PROVA/LADD/src/run.py", line 15, in run_experiment
    trainer = trainer_class(args, writer, device, rank, world_size)
  File "/home/utente/Scrivania/PROVA/LADD/src/federated/trainers/oracle_trainer.py", line 8, in __init__
    super().__init__(args, writer, device, rank, world_size)
  File "/home/utente/Scrivania/PROVA/LADD/src/federated/trainers/trainer.py", line 11, in __init__
    super().__init__(args, writer, device, rank, world_size)
  File "/home/utente/Scrivania/PROVA/LADD/src/general_trainer.py", line 58, in __init__
    self.__clients_setup()
  File "/home/utente/Scrivania/PROVA/LADD/src/general_trainer.py", line 143, in __clients_setup
    cl = client_class(**cl_args, batch_size=batch_size, test_user=split == 'test')
  File "/home/utente/Scrivania/PROVA/LADD/src/clients/oracle_client.py", line 16, in __init__
    super().__init__(args, client_id, dataset, model, writer, batch_size, world_size, rank, num_gpu,
  File "/home/utente/Scrivania/PROVA/LADD/src/clients/client.py", line 65, in __init__
    self.profiler_path = os.path.join('profiler', self.args.profiler_folder) if self.args.profiler_folder else None
  File "/home/utente/Scrivania/PROVA/LADD/src/clients/client.py", line 65, in __init__
    self.profiler_path = os.path.join('profiler', self.args.profiler_folder) if self.args.profiler_folder else None
  File "_pydevd_bundle/pydevd_cython.pyx", line 1589, in _pydevd_bundle.pydevd_cython.ThreadTracer.__call__
  File "_pydevd_bundle/pydevd_cython.pyx", line 929, in _pydevd_bundle.pydevd_cython.PyDBFrame.trace_dispatch
  File "_pydevd_bundle/pydevd_cython.pyx", line 920, in _pydevd_bundle.pydevd_cython.PyDBFrame.trace_dispatch
  File "_pydevd_bundle/pydevd_cython.pyx", line 317, in _pydevd_bundle.pydevd_cython.PyDBFrame.do_wait_suspend
  File "/home/utente/pycharm/plugins/python-ce/helpers/pydev/pydevd.py", line 1160, in do_wait_suspend
    self._do_wait_suspend(thread, frame, event, arg, suspend_type, from_this_thread)
  File "/home/utente/pycharm/plugins/python-ce/helpers/pydev/pydevd.py", line 1175, in _do_wait_suspend
    time.sleep(0.01)
KeyboardInterrupt